app/models/customer.rb

```diff
@@ -2,9 +2,8 @@ class Customer < ActiveRecord::Base
   has_many :invoices
   has_many :transactions, through: :invoices
 
-  def self.favorite_merchant(id)
-    customer = Customer.find(id)
-    merchant_id = customer.invoices.successful_transactions.group_by(&:merchant_id).max_by { |k, v| v.count }.flatten.first
+  def favorite_merchant
+    merchant_id = invoices.successful_transactions.group_by(&:merchant_id).max_by { |k, v| v.count }.flatten.first
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Line is too long. [114/80]